### ISSUE TITLE
How to export YOLOE model after linear probing?

### ISSUE THREAD

**Issue by arthur0219** [ultralytics/ultralytics#24777]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

This is the official example for exporting original yoloe-26l-seg model. After linear probing, text embedding is already fused into conv layer, it's no longer needed to "set_classes", then how to export the model?

```python

``` from ultralytics import YOLOE

# Select yoloe-26s/m-seg.pt for different sizes
model = YOLOE("yoloe-26l-seg.pt")

# Configure the set_classes() before exporting the model
model.set_classes(["person", "bus"])

export_model = model.export(format="onnx")
model = YOLOE(export_model)

# Run detection on the given image
results = model.predict("path/to/image.jpg")

# Show results
results[0].show()

### Additional

_No response_

**Comment by UltralyticsAssistant** [comment]
👋 Hello @arthur0219, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your question routed quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
Since this is a custom usage/export ❓ question about YOLOE after linear probing, please provide as much information as possible to help us investigate, including:
- The exact `ultralytics` version you are using
- Your full export code snippet
- Whether the model was only linear probed or additionally fine-tuned
- Any error messages or unexpected behavior during export/inference
- The target deployment runtime you plan to use with the exported ONNX model
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

You can export like normal YOLO model without `set_classes()` with `model.export()` and then load the model using `YOLO` instead  of `YOLOE`

```python

model.export(format="onnx")

model = YOLO("best.onnx")
```

**Comment by arthur0219** [comment]
Thank you for your reply.
By the way, if I only need yoloe text-prompt detection part, how to strip savpe, segmentation branch and all the other weights irrelevant to text-prompt detection? And when should I do this strip, before or after linear probing?

**Comment by glenn-jocher** [comment]
If you only need text-prompt detection, start from the matching detection YAML such as `yoloe-26l.yaml`, load `yoloe-26l-seg.pt`, and use `YOLOEPETrainer` rather than `YOLOEPESegTrainer`; that is the intended detection path, and `YOLOEPETrainer.get_model()` already removes `model.model[-1].savpe`, so do this before linear probing instead of pruning a `-seg.pt` model afterward—see [YOLOE detection training](https://docs.ultralytics.com/models/yoloe) and the [`YOLOEPETrainer` reference](https://docs.ultralytics.com/reference/models/yolo/yoloe/train).

**Comment by arthur0219** [comment]
Thanks a lot. @glenn-jocher 
When fine-tuning for object detection using the official fine-tuning code below, does the training process never update any parameters of RepRTA (SwiGLUFFN) itself? Instead, does it simply bake the enhanced text features output by RepRTA into the convolutional weights and only update those convolutional weights? If so, why not fine-tune SwiGLUFFN as well?

```python

``` from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer

# Initialize a detection model from a config
model = YOLOE("yoloe-26s.yaml")

# Load weights from a pretrained segmentation checkpoint (same scale)
model.load("yoloe-26s-seg.pt")

# Fine-tune on your detection dataset
results = model.train(
    data="coco128.yaml",  # Detection dataset
    epochs=80,
    patience=10,
    trainer=YOLOEPETrainer,  # <- Important: use detection trainer
)

**Comment by glenn-jocher** [comment]
Yes—under `YOLOEPETrainer`, `RepRTA`/`SwiGLUFFN` is not fine-tuned: `get_text_pe()` produces the text features, `set_classes()` + `fuse()` bake them into the detection head, and the trainer only re-enables the final `cv3[*][2]` conv layers for learning; that linear-probing design is intentional to keep adaptation cheaper and more stable while preserving pretrained text-image alignment, as shown in the [`YOLOEPETrainer.get_model()`](https://docs.ultralytics.com/reference/models/yolo/yoloe/train#ultralytics.models.yolo.yoloe.train.YOLOEPETrainer.get_model) and [`YOLOEDetect.fuse()`](https://docs.ultralytics.com/reference/nn/modules/head#ultralytics.nn.modules.head.YOLOEDetect.fuse) references.
